internal/provider/provider_test.go

```diff
@@ -44,26 +53,173 @@ var testAccProtoV6ProviderFactories = map[string]func() (tfprotov6.ProviderServe
+		case "<":
+			comparisonResult = auxillaryFunctions.IsVersionLesser(propertyV, apicV)
+		case ">=":
+			comparisonResult = auxillaryFunctions.IsVersionGreaterOrEqual(propertyV, apicV)
+		case "<=":
+			comparisonResult = auxillaryFunctions.IsVersionLessOrEqual(propertyV, apicV)
+		default:
+			return fmt.Errorf("Invalid operator found in the test check function: %s", operator)
+		}
+
+		if comparisonResult {
+			t.Logf("[WARNING] Test for checking the resource attribute value skipped due to version incompatibility between Property version: %s and APIC version: %s", propertyVersion, apicVersion)
+			return nil
+		}
+
+		for i, check := range checks {
+			if err := check(s); err != nil {
+				result = append(result, fmt.Errorf("Check %d/%d error: %w", i+1, len(checks), err))
+			}
+
+		}
+		return errors.Join(result...)
 	}
-	if v := os.Getenv("ACI_PASSWORD"); v == "" {
-		t.Fatal("ACI_PASSWORD must be set for acceptance tests")
+}
+
+func extractControllerVersion(requestData *container.Container) string {
+	classReadInfo := requestData.Search("imdata").Search("firmwareCtrlrRunning").Data().([]interface{})
+	if len(classReadInfo) == 1 {
+		attributes := classReadInfo[0].(map[string]interface{})["attributes"].(map[string]interface{})
+		for attributeName, attributeValue := range attributes {
+			if attributeName == "version" {
+				return attributeValue.(string)
+			}
+		}
 	}
-	if v := os.Getenv("ACI_URL"); v == "" {
-		t.Fatal("ACI_URL must be set for acceptance tests")
+	return ""
+}
+
+func testCheckResourceAttr(resourceName, attribute, value1 string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("resource not found: %s", resourceName)
+		}
+
+		attrValue, ok := rs.Primary.Attributes[attribute]
+		if !ok {
+			return nil
+		}
+
+		if attrValue != value1 {
+			return fmt.Errorf("attribute %s in resource %s should be %s , but got %s", attribute, resourceName, value1, attrValue)
+		}
+
+		return nil
 	}
-	if v := os.Getenv("ACI_VAL_REL_DN"); v == "" {
-		t.Fatal("ACI_VAL_REL_DN must be set for acceptance tests")
-		boolValue, err := strconv.ParseBool(v)
-		if err != nil || boolValue == true {
-			t.Fatal("ACI_VAL_REL_DN must be a 'false' boolean value")
+}
+
+func testCheckResourceDestroy(s *terraform.State) error {
+	aciClient := getACIClientTest(nil)
+
+	for name, rs := range s.RootModule().Resources {
+		if strings.HasPrefix(name, "aci_") {
+			_, err := aciClient.Get(rs.Primary.ID)
+			if err != nil {
+				if strings.Contains(err.Error(), "Error retrieving Object: Object may not exist") {
+					continue
+				} else {
+					return fmt.Errorf("error checking if resource '%s' with ID '%s' still exists: %s", rs.Type, rs.Primary.ID, err)
+				}
+			}
+			return fmt.Errorf("terraform destroy was unsuccessful. The resource '%s' with ID '%s' still exists", rs.Type, rs.Primary.ID)
```

Comment:
Can we add the resource variable name in this log?

Reply:
I replaced rs.Type with name which has both the type and name of the resource.